ledger/catchpointtracker.go

```diff
@@ -1593,16 +1593,51 @@ func (ct *catchpointTracker) accountsInitializeHashes(ctx context.Context, tx *s
 		// if anything goes wrong, it will still get rolled back.
 		_, err = trie.Evict(true)
 		if err != nil {
-			return fmt.Errorf("accountsInitialize was unable to commit changes to trie: %v", err)
+			return fmt.Errorf("initializeHashes was unable to commit changes to trie: %v", err)
+		}
+
+		// Now add the kvstore hashes
+		kvs, err := tx.QueryContext(ctx, "SELECT key, value FROM kvstore")
+		if err != nil {
+			return err
+		}
+		defer kvs.Close()
+		for kvs.Next() {
+			var k []byte
+			var v []byte
+			err := kvs.Scan(&k, &v)
+			if err != nil {
+				return err
+			}
+			hash := kvHashBuilderV6(string(k), v)
+			added, err := trie.Add(hash)
+			if err != nil {
+				return fmt.Errorf("initializeHashes was unable to add kv (key=%s) to trie: %v", k, err)
+			}
+			if !added {
+				if err != nil {
+					ct.log.Warnf("initializeHashes attempted to add duplicate kv hash '%s' to merkle trie for key %s : %v", hex.EncodeToString(hash), k, err)
+				} else {
+					ct.log.Warnf("initializeHashes attempted to add duplicate kv hash '%s' to merkle trie for key %s", hex.EncodeToString(hash), k)
+				}
```

Comment:
my point here is if err != nil the code exist at line 1615, not here. No need to have if-else here at all

Reply:
Good point! I copied it from above, which also has this problem, without thinking. Will fix both.

Reply:
the other use is fine since it looks up to resolve addrid to addr for a better log message

Reply:
true, my mistake